README.md

```diff
@@ -73,6 +73,37 @@ or
 import { Web5 } from CDN_LINK_HERE;
 ```
 
+### Additional Steps
+
+This package has transitive dependency on the [`@noble/ed25519`](https://github.com/paulmillr/noble-ed25519#usage) and [`@noble/secp256k1`](https://github.com/paulmillr/noble-secp256k1#usage) v2, additional steps are needed for some environments:
+
+- Node.js <= 18
+
+```js
+// node.js 18 and earlier,  needs globalThis.crypto polyfill
+import { webcrypto } from 'node:crypto';
+// @ts-ignore
+if (!globalThis.crypto) globalThis.crypto = webcrypto;
+```
+
+- React Native:
+
+```js
+// If you're on react native. React Native needs crypto.getRandomValues polyfill and sha512
+import "react-native-get-random-values";
+import { hmac } from "@noble/hashes/hmac";
+import { sha256 } from "@noble/hashes/sha256";
+import { sha512 } from "@noble/hashes/sha512";
+ed.etc.sha512Sync = (...m) => sha512(ed.etc.concatBytes(...m));
+ed.etc.sha512Async = (...m) => Promise.resolve(ed.etc.sha512Sync(...m));
+
+secp.etc.hmacSha256Sync = (k, ...m) =>
+  hmac(sha256, k, secp.etc.concatBytes(...m));
+secp.etc.hmacSha256Async = (k, ...m) =>
+  Promise.resolve(secp.etc.hmacSha256Sync(k, ...m));
+```
+
+
```

Comment:
@thehenrytsai

Why would these lines be suggested and how can they even be useful if we don't refer to them anywhere in the Web5 JS code base?

```javascript
ed.etc.sha512Sync = (...m) => sha512(ed.etc.concatBytes(...m));
ed.etc.sha512Async = (...m) => Promise.resolve(ed.etc.sha512Sync(...m));

secp.etc.hmacSha256Sync = (k, ...m) =>
  hmac(sha256, k, secp.etc.concatBytes(...m));
secp.etc.hmacSha256Async = (k, ...m) =>
  Promise.resolve(secp.etc.hmacSha256Sync(k, ...m));
  ```

It seems like this might only confuse people as they are going to get errors saying `ed` is `undefined` and `secp` is `undefined` unless they happen to have manually imported the library on a particular page within their app.

Reply:
> how can they even be useful to an app developer building with the SDK if we don't refer to them anywhere in the Web5 JS code base

Both @noble/secp256k1 and @noble/ed25519 are transitive dependencies of this repo:
<img width="701" alt="Screenshot 2023-07-01 at 12 48 27 PM" src="https://github.com/TBD54566975/web5-js/assets/3495974/a2360632-7bfc-4bf1-82d5-6f7c52223ed7">

> Why would these lines be suggested

These lines are prefaced by "React Native" on line 89 of the PR. They are listed under "React Native" setup steps by the author here: https://github.com/paulmillr/noble-ed25519#usage. According to the author, and confirmed locally by me, these lines are a necessity for using @noble/secp256k1 and @noble/ed25519 on certain runtimes. The callout in the README is intended to forewarn users about this. 

I've also created an automated setup: https://github.com/TBD54566975/web5-react-native-polyfills.

> It would only work if they happen to be attempting to write their own code using @noble/secp256k1 or @noble/ed25519 and had manually imported those libraries with import * as ed from '@noble/ed25519 and import * as secp from '@noble/secp256k1 (which btw, we don't even import using those names in the SDK).

This is not the case. Transitive dependencies are importable.

If it is unattractive to disclose setup steps for alternate runtimes on the front page of our docs may I suggest creating more pages: e.g. ALTERNATE-RUNTIMES.md and then we can drill down into these steps on those pages.

Reply:
@shamilovtim I don't think that really addresses the questions.

We don't use any of these functions anywhere in the Web5 packages:
`ed.etc.sha512Sync`
`ed.etc.sha512Async`
`secp.etc.hmacSha256Sync`
`secp.etc.hmacSha256Async`

Sure, I understand if a developer did use these functions, they need to add the lines to your project, but Web5 JS does not use them currently.  Why are you having to polyfill in React Native for functions that we don't use?

Reply:
Confirmed locally that the package author's lines of code are required: 

https://github.com/TBD54566975/web5-js/assets/3495974/345330bd-4797-417a-80f6-59402971ba01





> Why are you having to polyfill in React Native for functions that we don't use?

This is the the global setup of the `@noble/ed25519` and `@noble/secp256k1` packages as [required](https://github.com/paulmillr/noble-ed25519#usage) by the maintainer of those packages. Based on his README, he isn't saying "only use these polyfills if you plan to use certain functions." He's saying:

>  React Native needs [...] sha512

![Screenshot 2023-07-03 at 12 38 46 AM](https://github.com/TBD54566975/web5-js/assets/3495974/bec18ee8-61ae-4ca9-ac4a-8363c2a9cf34)